src/modules/credentials/handlers/IssueCredentialHandler.ts

```diff
@@ -1,17 +1,58 @@
+import type { AgentConfig } from '../../../agent/AgentConfig'
 import type { Handler, HandlerInboundMessage } from '../../../agent/Handler'
+import type { CredentialRecord } from '../repository'
 import type { CredentialService } from '../services'
 
+import { createOutboundMessage } from '../../../agent/helpers'
+import { AriesFrameworkError } from '../../../error/AriesFrameworkError'
+import { AutoAcceptCredential } from '../../../types'
+import { CredentialUtils } from '../CredentialUtils'
 import { IssueCredentialMessage } from '../messages'
 
 export class IssueCredentialHandler implements Handler {
   private credentialService: CredentialService
+  private agentConfig: AgentConfig
   public supportedMessages = [IssueCredentialMessage]
 
-  public constructor(credentialService: CredentialService) {
+  public constructor(credentialService: CredentialService, agentConfig: AgentConfig) {
     this.credentialService = credentialService
+    this.agentConfig = agentConfig
   }
 
   public async handle(messageContext: HandlerInboundMessage<IssueCredentialHandler>) {
-    await this.credentialService.processCredential(messageContext)
+    const credentialRecord = await this.credentialService.processCredential(messageContext)
+
+    const autoAccept = CredentialUtils.composeAutoAccept(
+      credentialRecord.autoAcceptCredential,
+      this.agentConfig.autoAcceptCredentials
+    )
+
+    if (autoAccept === AutoAcceptCredential.always) {
+      return await this.nextStep(credentialRecord, messageContext)
+    } else if (autoAccept === AutoAcceptCredential.contentApproved) {
+      if (credentialRecord.credentialAttributes && credentialRecord.credentialMessage) {
+        const indyCredential = credentialRecord.credentialMessage.indyCredential
+
+        if (!indyCredential) {
+          throw new AriesFrameworkError(`Missing required base64 encoded attachment data for credential`)
```

Comment:
I don't think we should throw an error here, but rather do an error log (this.logger.error). As we can't do anything with the error on a higher level (it is not invoked by the user, but by another agent.)